### PR TITLE
Adds category selector for the fill component

### PR DESCRIPTION
**Pull request by javierarce**
Pending stuff: 

- Add selected state for the color ramps.
- Show the 'Others' category at the bottom of the category list.
- Select a default ramp when switching to the ramp view.
- Add specs for the category list

### CODE REVIEW

**Comment by javierarce** [comment]
CR: 🙏  @xavijam

File: app/assets/stylesheets/editor-3/carousel.css.scss [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Styles for carousel (icons active and normal)
```
Comment (xavijam) [inline app/assets/stylesheets/editor-3/carousel.css.scss]: I can't comment the file above so 👏  👏

File: lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-collection.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ module.exports = Backbone.Collection.extend({
 
   initialize: function () {
     this.bind('change:selected', this._onSelectedChange, this);
-    this.bind('change change:range', this._onModelsChanged, this);
+    this.bind('change change:domain change:range', this._onModelsChanged, this);
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-collection.js]: if we have `change`, do we need rest of them?

File: lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-categories/input-color-categories-list-view.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  initialize: function (opts) {
+    this._showSearch = opts.showSearch || false;
+    this._typeLabel = opts.typeLabel;
+
+    this._collection = new CustomListCollection();
+    this._initBinds();
+  },
+
+  render: function () {
+    this._setupCollection();
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-categories/input-color-categories-list-view.js]: I'd avoid any setup in the render method.

File: lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-categories/input-color-categories-list-view.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+  _onClickBack: function (e) {
+    this.killEvent(e);
+    this.trigger('back', this);
+  },
+
+  _initBinds: function () {
+    this.model.on('change:range', this.render, this);
+    this.model.on('change:domain', this.render, this);
+    this._collection.bind('change:selected', this._onSelectItem, this);
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-categories/input-color-categories-list-view.js]: this.add_related_model(this._collection);

Why do we use `_collection`.

File: lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-categories/input-color-categories.js [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+      createStackView: function (stackLayoutModel, opts) {
+        return self._createRangeListView(stackLayoutModel, opts).bind(self);
+      }
+    }, {
+      createStackView: function (stackLayoutModel, opts) {
+        return self._createColorPickerView(stackLayoutModel, opts).bind(self);
+      }
+    }]);
+
+    this._stackLayoutView = new StackLayoutView({
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-categories/input-color-categories.js]: Don't we need to add this view as subview? `this.addView(this._stackLayoutView)`

File: vendor/assets/javascripts/cartodb.uncompressed.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
 // cartodb.js version: 3.15.9
```
Comment (xavijam) [inline vendor/assets/javascripts/cartodb.uncompressed.js]: Do we need to update cartodb.js?

Comment (xavijam) [inline vendor/assets/javascripts/cartodb.uncompressed.js]: Also oddyssey has been removed 😱 .
It is likely we need an update with master.

File: lib/assets/javascripts/cdb [hunk 7]
```diff
@@ -1 +1 @@
-Subproject commit ea0290f51d64a2772c69af6ef67dfa7e9af7a9d0
```
Comment (xavijam) [inline lib/assets/javascripts/cdb]: Not needed, I hope.

File: lib/assets/javascripts/cartodb3/editor/style/style-form/style-aggregation-form/style-aggregation-form-view.js [hunk 8]
```diff
@@ -8,8 +8,10 @@ module.exports = cdb.core.View.extend({
 
   initialize: function (opts) {
     if (!opts.querySchemaModel) throw new Error('querySchemaModel is required');
+    if (!opts.configModel) throw new Error('configModel is required');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/style/style-form/style-aggregation-form/style-aggregation-form-view.js]: I feel you don't need the configModel for the aggregation form. It doesn't render any fill or stroke component.

File: lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-number/input-number-value-content-view.js [hunk 9]
```diff
@@ -4,7 +4,7 @@ var StackLayoutView = require('../../../../../components/stack-layout/stack-layo
 var ColumnListView = require('../column-list-view');
 var InputNumberContentView = require('./input-number-content-view');
 
-var QUANTIFICATION_METHODS = ['Jenks', 'Equal Interval', 'Heads/Tails', 'Quantile']; // TODO: translate?
+var QUANTIFICATION_METHODS = ['Jenks', 'Equal Interval', 'Heads/Tails', 'Quantiles']; // TODO: translate?
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-number/input-number-value-content-view.js]: Nice! thanks, could we translated them BTW?

File: lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-ramps/input-color-ramps.js [hunk 10]
```diff
@@ -0,0 +1,132 @@
+var _ = require('underscore');
+var cdb = require('cartodb.js');
+var Backbone = require('backbone');
+var StackLayoutView = require('../../../../../../components/stack-layout/stack-layout-view');
+var ColumnListView = require('../../column-list-view');
+var InputRampContentView = require('./input-ramp-content-view');
+var rampList = require('./ramps');
+
+var QUANTIFICATION_METHODS = ['Jenks', 'Equal Interval', 'Heads/Tails', 'Quantiles']; // TODO: translate
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-ramps/input-color-ramps.js]: I've seen those static variables two times, could we move somewhere and start using them from there?

**Comment by xavijam** [comment]
The things I've just commented, but 👍 .

File: vendor/assets/javascripts/cartodb.uncompressed.js (hunk 6 above)
Comment (javierarce) [inline vendor/assets/javascripts/cartodb.uncompressed.js]: I think I added the file inadvertently :(

**Comment by javierarce** [comment]
Reopened [here](https://github.com/CartoDB/cartodb/pull/7642) with comments addressed.

**Pull request closed without merging**